lib/galaxy/tools/wrappers.py

```diff
@@ -280,6 +283,15 @@ def is_of_type(self, *exts):
                 log.warning("Datatype class not found for extension '%s', which is used as parameter of 'is_of_type()' method" % (e))
         return self.dataset.datatype.matches_any(datatypes)
 
+    @property
+    def safe_element_identifier(self):
+        """
+        creates a string from the element identifier that can be used safely as filename
+        by replacing the the only character that is forbidden in bash script (the slash)
+        with an underscore.
+        """
+        return self.element_identifier.replace("/","_")
```

Comment:
There's more than just that, `;`, `"`, `'`, `$`, `(`, `)` and plenty more that shouldn't be allowed. For a first try I'd probably go with `valid_filename_chars` in galaxy.util

Reply:
Actually I have used `sanitize_for_filename` before implementing this less intrusive version (therefore the unnecessary import .. @bgruening).